### PR TITLE
check last update job monitor and if more than a timeout, re-submit

### PR DESCRIPTION
**Pull request by burnout87**

### CODE REVIEW

**Comment by burnout87** [comment]
@volodymyrss not sure the test I wrote covers enough of the cases where a re-submission should happen.

What other cases should we cover?

File: cdci_data_analysis/config_dir/conf_env.yml.example [hunk 1]
```diff
@@ -31,6 +31,9 @@ dispatcher:
     # maximum interval allowed during token refreshing
     token_max_refresh_interval: 604800
 
+    # timeout to re-submit the request
+    resubmit_timeout: 1800
```
Comment (volodymyrss) [inline cdci_data_analysis/config_dir/conf_env.yml.example:35]: Just that this makes sense to me

```suggestion
    resubmit_timeout: 900
```

**Comment by volodymyrss** [comment]
indeed, we may want to add something like:
1. set 1s resumit threshold, add somewhere in dummy a marker/log that request is being sent to backend
2. submit a job and check the marker that the request WAS sent to backend
3. do run_anaysis again and check the marker that the request was NOT sent to backend
4. sleep 1.5s
5. do run_anaysis again and check the marker that the request WAS sent to backend

**Comment by burnout87** [comment]
> 

test added

I included also a case where the job, on the dummy backend, is marked as `done`. Then the `run_analysis` is re-submitted, getting a `ready` as a job_status

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 2]
```diff
@@ -1735,7 +1770,64 @@ def run_query(self, off_line=False, disp_conf=None):
...
+                        query_out = self.instrument_run_query(product_type,
+                                                              job,
+                                                              run_asynch,
+                                                              query_type,
+                                                              verbose,
+                                                              dry_run,
+                                                              api)
+                        self.log_query_progression("after re-submission of instrument.run_query")
+                    except RequestNotAuthorized as e:
+                        # TODO why is it an oda_api related response ?
```
Comment (volodymyrss) [inline cdci_data_analysis/flask_app/dispatcher_query.py:1792]: it should not be, please change

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 3]
```diff
@@ -1788,23 +1874,16 @@ def run_query(self, off_line=False, disp_conf=None):
...
-                                                          dry_run=dry_run,
-                                                          api=api,
-                                                          decoded_token=self.decoded_token)
+                                                          query_type,
+                                                          verbose,
+                                                          dry_run,
+                                                          api)
                     self.log_query_progression("after instrument.run_query")                                                          
                 except RequestNotAuthorized as e:
+                    # TODO why is it an oda_api related response ?
```
Comment (volodymyrss) [inline cdci_data_analysis/flask_app/dispatcher_query.py:1886]: why indeed

**Comment by burnout87** [comment]
All tests passed.

**Review by volodymyrss — APPROVED** [review]
I looked as much as I could at this time.

**Comment by burnout87** [comment]
All test are now passing, but not much as changed since the last review, so, it can be merged if you thinks it's ok @volodymyrss  @dsavchenko

**Review by volodymyrss — APPROVED** [review]

**Review by dsavchenko — APPROVED** [review]

**Pull request merged**